### DISCUSSION TITLE
public directory is killing me...

### DISCUSSION THREAD

**Discussion by szarkans** [vuejs/vitepress#4367] [Q&A] [answered]
Vitepress just can't find any files in `public` directory, even though folder is in the same folder as `.vitepress`. I just don't know what to do, i tried everything.
I'm trying to add favicon to pages and image for index page, but i just can't. It driving me insane...

`head: [['link', { rel: 'icon',  type: 'image/png', href: '/icon.png' }]],`

![image](https://github.com/user-attachments/assets/2c78e546-a8b3-4639-8459-f02acbacba64)

**Comment by szarkans** [comment]
![image](https://github.com/user-attachments/assets/cc4d7730-a34f-4cee-b138-7d30013dd1f4)

i cant take it anymore. i even tried to add `/public/` to href, but still nothing.

**Comment by szarkans** [comment, marked as answer]
Kinda found it...
Had to set `base: '/'` in `config.mjs`, and VitePress accepts favicon if file name is `favicon`

**Reply by brc-dd** [reply]
if you don't want `/` as base, you should change that href to `/<your-base>/icon.png`

> VitePress accepts favicon if file name is favicon

this isn't vitepress specific. if you've a file favicon.ico accessible at `/favicon.ico` the browsers will automatically pull it even if you don't have a specific link tag.

**Reply by szarkans** [reply]
> if you don't want `/` as base, you should change that href to `//icon.png


I'm not really sure how to keep every `.md` file in `docs` folder and having `./` as project root and don't use `domain.com/docs/`
Keeping all folders in one folder is kinda annoying.
